src/invoice/signature.rs

```diff
@@ -224,6 +346,31 @@ impl SecretKeyFile {
     }
 }
 
+impl SecretKeyStorage for SecretKeyFile {
+    fn get_first_matching(&self, role: &SignatureRole) -> Option<&SecretKeyEntry> {
+        self.key.iter().find(|k| k.roles.contains(role)).clone()
+    }
+}
+
+pub struct SecretKeyFileLoader {
+    path: PathBuf,
```

Comment:
This is a private data member but I don't see a `new` or other type of constructor here for someone using it publicly

Reply:
Removed